src/components/pin/pin.repository.ts

```diff
@@ -22,28 +22,29 @@ export class PinRepository {
     return result ? true : false;
   }
 
-  async togglePinned(
-    id: string,
-    pinned: boolean,
-    session: Session
-  ): Promise<void> {
-    await this.removeOldPinned(id, session);
-    if (pinned) {
-      const createdAt = DateTime.local();
-      await this.db
-        .query()
-        .call(matchRequestingUser, session)
-        .match([node('node', 'BaseNode', { id })])
-        .create([
-          node('requestingUser'),
-          relation('out', '', 'pinned', { active: true, createdAt }),
-          node('node'),
-        ])
-        .run();
+  async add(id: string, session: Session): Promise<void> {
+    const isPinned = await this.isPinned(id, session);
+    if (isPinned) {
+      return;
     }
+    const createdAt = DateTime.local();
+    await this.db
+      .query()
+      .call(matchRequestingUser, session)
+      .match([node('node', 'BaseNode', { id })])
+      .create([
```

Comment:
Maybe `createUnique()` would work here to prevent duplicates while not having to check is pinned?

Reply:
`createUnique` is deprecated. I think it will be better to check if it's already pinned or unpinned in the service before the action.

Reply:
Oh right. I think we want `merge` here.